src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Specs/Types/ObjectSpec.cs

```diff
@@ -14,13 +13,13 @@ internal sealed record ObjectSpec : ComplexTypeSpec
 
         public override TypeSpecKind SpecKind => TypeSpecKind.Object;
 
-        public override bool HasBindableMembers => Properties.Values.Any(p => p.ShouldBindTo);
+        public override bool HasBindableMembers => Properties?.Any(p => p.ShouldBindTo) is true;
 
         public override bool CanInstantiate => InstantiationStrategy is not InstantiationStrategy.None && InitExceptionMessage is null;
 
-        public Dictionary<string, PropertySpec> Properties { get; } = new(StringComparer.OrdinalIgnoreCase);
+        public ImmutableEquatableArray<PropertySpec>? Properties { get; set; }
 
-        public List<ParameterSpec> ConstructorParameters { get; } = new();
+        public ImmutableEquatableArray<ParameterSpec>? ConstructorParameters { get; set; }
```

Comment:
Why are these `set` members and not `init`? When they are `set` then it's basically a mutable value. That's not wrong but it does open the door to a class of bugs. Namely mutating a value that say the incremental generator is caching and believes is unchanging.